Automattic/wp-calypso
# Components: Reduxify the <Shortcode /> component

This PR reduxifies the `<Shortcode />` component and implements the related media flux actions. It depends on #8116, which is already in the wild.

To do in this PR:
- [x] Implement into the main <Shortcode /> component.
- [x] Remove the legacy shortcode data component.
- [x] Update `ShortcodeFrame` to use `result` instead of `body`.
- [x] ~~Implement the (still) Flux media actions `RECEIVE_MEDIA_ITEMS` and `RECEIVE_MEDIA_ITEM` with the Redux shortcode reducer.~~
- [ ] Prepare PR for integrating Flux media actions `RECEIVE_MEDIA_ITEMS` and `RECEIVE_MEDIA_ITEM` with  #2167.
- [x] Remove `lib/shortcodes`.
- [x] Test all affected components manually.

To test:

* Checkout this branch
* Verify there are no regressions within:
  * Post editor shortcodes (rendering and editing items, captions, inserting, reordering and deleting items)
  * Post editor media modal (editing media item data, including title and secondary fields)
  * Post editor featured image (rendering and editing)
  * Media library - dropping a file, uploading a file through URL, uploading a file directly
* Verify all tests pass
